src/musig.rs

```diff
@@ -0,0 +1,1126 @@
+//! This module implements high-level Rust bindings for a Schnorr-based
+//! multi-signature scheme called MuSig2 [paper](https://eprint.iacr.org/2020/1261).
+//! It is compatible with bip-schnorr.
+//!
+//! The documentation in this module is for reference and may not be sufficient
+//! for advanced use-cases. A full description of the C API usage along with security considerations
+//! can be found in [C-musig.md](secp256k1-sys/depend/secp256k1/src/modules/musig/musig.md).
+use core;
+use core::fmt;
+use core::mem::MaybeUninit;
+#[cfg(feature = "std")]
+use std;
+
+use crate::ffi::{self, CPtr};
+use crate::{
+    schnorr, Error, Keypair, Message, PublicKey, Scalar, Secp256k1, SecretKey, Signing, Verification, XOnlyPublicKey
+};
+
+/// Musig partial signature parsing errors
+#[derive(Debug, Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Hash)]
+pub enum ParseError {
+    /// Parse Argument is malformed. This might occur if the point is on the secp order,
+    /// or if the secp scalar is outside of group order
+    MalformedArg,
+}
```

Comment:
Wouldn't it make more sense to use Option in the `from_byte_array` functions? The underlying functions do not provide information about why they failed anyway, so this type is not too useful (unless it is expected to be expanded in the future, in which case it should be `#[non_exhaustive]`).

Reply:
Even if we didn't intend to expand it, an empty error type is often better than `Option` (which is the case here since passing malformed argument is clearly an error as opposed to stuff like passing non-existing key to `get` function on a map).